frontend/src/core/cells/cells.ts

```diff
@@ -214,17 +216,24 @@ const {
         : state.cellIds.topLevelIds.indexOf(cellId);
     const insertionIndex = before ? index : index + 1;
 
+    let cellContents = code;
+    const cellEditorView = getCellEditorView(cellId as CellId);
```

Comment:
i think we should pass a param to `createNewCell` whether we should `includeSelectionAsInitialCode`. 

we run `createNewCell` in many places and this would be weird for most cases (e.g. snippets, data-source panel, AI cell, etc).

could we also assert `code` and `includeSelectionAsInitialCode` are never both true?

Reply:
Little confused here: do we need `includeSelectionAsInitialCode` if we are already redefining the `code` prop?